Add Coherence state store component docs (#4696)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-coherence.md

---
type: docs
title: "Coherence"
linkTitle: "Coherence"
description: Detailed information on the Coherence state store component
aliases:
  - "/operations/components/setup-state-store/supported-state-stores/setup-coherence/"
---

## Component format

To setup Coherence state store create a component of type `state.coherence`. See [this guide]({{< ref "howto-get-save-state.md#step-1-setup-a-state-store" >}}) on how to create and apply a state store configuration.

```yaml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: <NAME>
spec:

>   type: state.coherence

Reviewer: This is an Oracle DB yes? Should this type then be `state.oracle.coherence` or `state.oci.coherence`

Author: This is not part of Oracle DB, and not dependant on it, but it is an Oracle product, for in memory caching in general.
In components-contrib i have already committed as:
```
  type: state.coherence
```

It is not sepcific to cloud like the Object Storage and ADW.

thoughts?

| tlsClientKey       |    N     | Client key for Coherence. Defaults to "". Can be `secretKeyRef` to use a [secret reference]({{< ref component-secrets.md >}}).              | `"-----BEGIN CERTIFICATE-----\nMIIC9TCCA..."` |
| tlsCertsPath       |    N     | Additional certificates for Coherence. Defaults to "". Can be `secretKeyRef` to use a [secret reference]({{< ref component-secrets.md >}}). | `"-----BEGIN CERTIFICATE-----\nMIIC9TCCA..."` |
| ignoreInvalidCerts |    N     | Indicates if to ignore self-signed certificates for testing only, not to be used in production. Defaults to false                           | `"false"`                                     |
| scopeName          |    N     | A scope name to use for the internal cache. Defaults to ""                                                                                  | `"my-scope"`                                  |
| requestTimeout     |    N     | ATimeout for calls to the cluster Defaults to "30s"                                                                                         | `"15s"`                                       |

> | nearCacheTTL       |    N     | If non zero a near cache will be used and the TTL of the near cache will be this value. Defaults to 0s                                      | `"60s"`                                       |

Reviewer: ```suggestion
| nearCacheTTL       |    N     | If non-zero a near cache is used and the TTL of the near cache is this value. Defaults to 0s                                      | `"60s"`                                       |
```

> | nearCacheUnits     |    N     | If non zero a near cache will be used and the maximum size of the near cache will be this value in units. Defaults to 0                     | `"1000"`                                      |

Reviewer: ```suggestion
| nearCacheUnits     |    N     | If non-zero a near cache is used and the maximum size of the near cache is this value in units. Defaults to 0                     | `"1000"`                                      |
```

> | nearCacheMemory    |    N     | If non zero a near cache will be used and the maximum size of the near cache will be this value in bytes. Defaults to 0                     | `"4096"`                                      |

Reviewer: ```suggestion
| nearCacheMemory    |    N     | If non-zero a near cache is used and the maximum size of the near cache is this value in bytes. Defaults to 0                     | `"4096"`                                      |
```

```bash
NAME                    TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)                                               AGE
kubernetes              ClusterIP   10.96.0.1      <none>        443/TCP                                               9m
my-cluster-grpc         ClusterIP   10.96.225.43   <none>        1408/TCP                                              7m3s
my-cluster-management   ClusterIP   10.96.41.6     <none>        30000/TCP                                             7m3s
my-cluster-sts          ClusterIP   None           <none>        7/TCP,7575/TCP,7574/TCP,6676/TCP,30000/TCP,1408/TCP   7m3s
my-cluster-wka          ClusterIP   None           <none>        7/TCP,7575/TCP,7574/TCP,6676/TCP                      7m3s
```

For example, if installing using the example above, the Coherence host address would be:

`my-cluster-grpc`
{{% /codetab %}}

{{< /tabs >}}

## Related links
- [Basic schema for a Dapr component]({{< ref component-schema >}})
- Read [this guide]({{< ref "howto-get-save-state.md#step-2-save-and-retrieve-a-single-state" >}}) for instructions on configuring state store components

> - [State management building block]({{< ref state-management >}})

Reviewer: Is there any useful reference to the Coherence docs that you would consider adding here?

Author: Of course, makes sense.
